src/main/java/it/rate/webapp/controllers/InterestAdminController.java

```diff
@@ -23,18 +25,25 @@
 public class InterestAdminController {
   private final InterestService interestService;
   private final ManageInterestService manageInterestService;
-
+  private final UserService userService;
 
   @GetMapping("/edit")
   @PreAuthorize("hasAnyAuthority(@permissionService.manageCommunity(#interestId))")
-  public String editInterestPage(@PathVariable Long interestId, Model model) {
+  public String editInterestPage(@PathVariable Long interestId, Model model, Principal principal) {
     Optional<Interest> interest = interestService.findInterestById(interestId);
     if (interest.isEmpty()) {
       throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Interest not found");
     }
     model.addAttribute("interest", interest.get());
     model.addAttribute("action", "/interests/" + interestId + "/admin/edit");
     model.addAttribute("method", "put");
+    if (principal != null) {
+      model.addAttribute(
+              "loggedUser",
+              userService
+                      .findByEmail(principal.getName())
+                      .orElseThrow(() -> new RuntimeException("Email not found in the database")));

```

Comment:
Use getByEmail without using the optional

Reply:
done